### PR TITLE
EID-1254: Generate and Sign Connector Metadata

### PR DESCRIPTION
**Pull request by dbes-gds**
Re-factoring of local/dev metadata generation scripts to create stand-alone connector metadata generation script that can be run from the command-line.
Some groundwork done to sign the metadata with the hsm, this will be completed in a subsequent task.
The stand-alone script currently does not sign the metadata at all, however the local/dev metadata generation and signing continues to work as-is.

Add separate command-line option to generate and sign connector metadata.

### CODE REVIEW

File: pki/proxy-node-metadata.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      xml.at_xpath("//md:EntityDescriptor//md:IDPSSODescriptor").add_previous_sibling(extensions)
+    end
+  end
+end
+
+#
+#   Sign Metadata locally - requires a keypair and an xmlsectool installation
+#
+def sign_metadata(metadata_xml, keypair, xmlsectool_path = nil)
+  xmlsectool_path ||= ENV.fetch('XMLSECTOOL', 'xmlsectool')
```
Comment (tomrosier) [inline pki/proxy-node-metadata.rb]: This check should go into `gen_sign_proxy_node_metadata` when the `xmlsectool_path` is extracted.

File: pki/generate [hunk 2]
```diff
@@ -6,6 +6,7 @@ require 'yaml'
 
 require_relative 'certs'
 require_relative 'metadata'
+require_relative 'proxy-node-metadata'
```
[the remark this answers is not in the document]
Comment (dbes-gds) [reply pki/generate]: The 'sign_metadata' method is in proxy-node-metadata, shared by both the hub and proxy-node local metadata generation.

[the remark this answers is not in the document]
Comment (tomrosier) [reply pki/generate]: Okay that makes sense 👍

File: pki/proxy-node-metadata.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        puts(YAML.dump(proxy_node_config) + "---")
+      end
+    end
+
+    generated_xml = capture_output do
+      Verify::Metadata::Generator.run!(%W(-e #{metadata_env} --valid-until=365 --proxy-node --idpCA ca.crt))
+    end
+
+    Nokogiri::XML(generated_xml).tap do |xml|
+      extensions = <<~EOS
```
[the remark this answers is not in the document]
Comment (chrisfarms) [reply pki/proxy-node-metadata.rb]: Can we get this into the upstream `Verify::Metadata::Generator`? it looks like it already has a `--proxy-node` arg and unless I'm mistaken this is essentially a constant that always needs to be appended whenever `--proxy-node` metadata is generated?

**Pull request merged**